prqlc/prqlc/tests/integration/sql.rs

```diff
@@ -4977,3 +4977,18 @@ fn test_param_declarations() {
       b = $a
     "###);
 }
+
+#[test]
+fn test_relation_aliasing() {
+    assert_display_snapshot!(compile(
+        r###"
+    from x | select {y = this} | select {y.hello}
```

Comment:
Cool! So `y` is a tuple / relation

Reply:
Idea is that if `x` is of type `[{a = int, b = bool}]`, then
```
x | select {y = this}
```
... will have the type of `[{y = {a = int, b = bool}}]`.

... which then gets flattened when converting to an SQL relation.

Reply:
Yes perfect.

Will be great to have an LSP one day, would be so cool to hover over `y` and get that type...